Yeah, running as root on a VM and I've got my PACKAGESITE set. It might be failing on the cleanup

```
ERunning cleanup command 'pkg delete -y sysutils/py-salt'
======================================================================
ERROR: test_install_daily (bootstrap.test_install.InstallationTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/salt-bootstrap/tests/bootstrap/test_install.py", line 139, in test_install_daily
    args=('daily',), timeout=15 * 60, stream_stds=True
  File "/root/salt-bootstrap/tests/bootstrap/__init__.py", line 138, in run_script
    while process.poll() is None:
  File "/root/salt-bootstrap/tests/bootstrap/__init__.py", line 74, in poll
    obuff = self.stdout.read()
IOError: [Errno 35] Resource temporarily unavailable
```

--- a/bootstrap-salt.sh
+++ b/bootstrap-salt.sh
@@ -10,6 +10,7 @@
 #          BUGS: https://github.com/saltstack/salty-vagrant/issues
 #        AUTHOR: Pedro Algarvio (s0undt3ch), pedro@algarvio.me
 #                Alec Koumjian (akoumjian), akoumjian@gmail.com
+#                Geoff Garside (geoffgarside), geoff@geoffgarside.co.uk
 #       LICENSE: Apache 2.0
 #  ORGANIZATION: Salt Stack (saltstack.org)
 #       CREATED: 10/15/2012 09:49:37 PM WEST

--- a/tests/bootstrap/__init__.py
+++ b/tests/bootstrap/__init__.py
@@ -75,7 +75,7 @@ class NonBlockingPopen(subprocess.Popen):
                 self.obuff += obuff
                 sys.stdout.write(obuff)
             except IOError, err:
-                if err.errno != 11:
+                if err.errno not in (11, 35):
                     # We only handle Resource not ready properly, any other
                     # raise the exception
                     raise
@@ -85,7 +85,7 @@ class NonBlockingPopen(subprocess.Popen):
                 self.ebuff += ebuff
                 sys.stderr.write(ebuff)
             except IOError, err:
-                if err.errno != 11:
+                if err.errno not in (11, 35):
                     # We only handle Resource not ready properly, any other
                     # raise the exception
                     raise
@@ -179,14 +179,17 @@ class BootstrapTestCase(TestCase):
                 # process already terminated
                 pass
 
-    def assert_script_result(self, fail_msg, expected_rc, process_details):
+    def assert_script_result(self, fail_msg, expected_rcs, process_details):
+        if not isinstance(expected_rcs, (tuple, list)):
+            expected_rcs = (expected_rcs,)
+
         rc, out, err = process_details
-        if rc != expected_rc:
+        if rc not in expected_rcs:
             err_msg = '{0}:\n'.format(fail_msg)
             if out:
                 err_msg = '{0}STDOUT:\n{1}\n'.format(err_msg, '\n'.join(out))
             if err:
                 err_msg = '{0}STDERR:\n{1}\n'.format(err_msg, '\n'.join(err))
             if not err and not out:
-                err_msg = '{0} No std{out,err} captured.'.format(err_msg)
+                err_msg = '{0} No stdout nor stderr captured.'.format(err_msg)
             raise AssertionError(err_msg.rstrip())

--- a/tests/bootstrap/test_install.py
+++ b/tests/bootstrap/test_install.py
@@ -47,7 +47,11 @@ class InstallationTestCase(BootstrapTestCase):
             print 'Running cleanup command {0!r}'.format(cleanup)
             self.assert_script_result(
                 'Failed to execute cleanup command {0!r}'.format(cleanup),
-                0,
+                (
+                    0,
+                    65  # FreeBSD throws this error code when the packages being
+                        # un-installed were not installed in the first place
+                ),
                 self.run_script(
                     script=None,
                     args=cleanup.split(),

--- a/tests/bootstrap/test_install.py
+++ b/tests/bootstrap/test_install.py
@@ -24,7 +24,7 @@ CLEANUP_COMMANDS_BY_OS_FAMILY = {
         'yum -y remove salt-minion salt-master'
     ],
     'FreeBSD': [
-        'pkg delete -y sysutils/py-salt',
+        'pkg delete -y swig sysutils/py-salt',
         'pkg autoremove -y'
     ]
 }

